pkg/virt-api/rest/subresource.go

```diff
@@ -165,6 +165,27 @@ func (app *SubresourceAPIApp) putRequestHandler(request *restful.Request, respon
 	}
 }
 
+func (app *SubresourceAPIApp) USBRedirRequestHandler(request *restful.Request, response *restful.Response) {
+	validate := func(vmi *v1.VirtualMachineInstance) *errors.StatusError {
+		condManager := controller.NewVirtualMachineInstanceConditionManager()
+		if condManager.HasCondition(vmi, v1.VirtualMachineInstancePaused) {
+			return errors.NewConflict(v1.Resource("virtualmachineinstance"), vmi.Name, fmt.Errorf("VMI is paused"))
```

Comment:
What happens if a VMI is paused after a device is connected?

Reply:
That's something I didn't test before, so I gave it a try this morning for a little while. If we are talking about storage, like flash disk, it works fine like `console` or `vnc` commands from `virtctl`, that is, it stays connected to the guest when I `unpause` it. I tried with vmi-alpine-efi.

With a isoc device, like a webcam, I've used the vmi-fedora plus the necessary kernel modules and gstreamer1 tools to fetch data from the device (using a fakevideosink, for this test) while I pause it. The `usbredirect` eventually crashed, that means that the guest would not have its device when the VMI is `unpaused`.

This was somehow handled in the past in the usbredir library, I've added a [callback in the library](https://gitlab.freedesktop.org/spice/usbredir/commit/a88e197b18785d6de2322b5f26484c4130a6f2b9) itself to drop isoc packets based on the network connection (e.g: Streaming high quality video when the network can't handle the bandwith).

That said, this seems to be a fix in the usbredirect itself and I'll work on that for its next release, I hope it doesn't block this PR :)

Reply:
Opened an issue for this [here](https://gitlab.freedesktop.org/spice/usbredir/-/issues/19)